test_server.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+import pytest
+from urllib2 import HTTPError
+import socket
+from http_server import response_ok, response_error, parse_request
+
+
+@pytest.fixture()
+def make_client():
+    ADDR = ("127.0.0.1", 8000)
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.connect(ADDR)
+    return client
+
+
+def helper(client, msg):
+    try:
+        client.sendall(msg)
+        client.shutdown(socket.SHUT_WR)
+        response = ""
+        while True:
+            part = client.recv(16)
+            response += part
+            if len(part) < 16:
+                client.close()
+                return response
+    except Exception as e:
+        print e
+
+
+def test_client1(make_client):
+    client = make_client
+    assert "HTTP/1.1 200 OK" in helper(client, "GET /path/templates/thing.html HTTP/1.1\r\nHOST: www.site.com\r\n\r\n")
+
+
+def test_response_ok():
+    assert "HTTP/1.1 200 OK" in response_ok()
```

Comment:
this test and the one below do not test that the string returned by `response_ok` is in fact an HTTP response.  The merely match strings.  This would pass too:

```python
assert "HTTP/1.1 200 ok' in "The proper formatting for an HTTP response has a first line like this: 'HTTP/1.1 200 OK'"
```
But that's clearly not a good HTTP response.  Make sure your tests test something real.